x/sync/peer_tracker.go

```diff
@@ -36,6 +36,18 @@ const (
 	randomPeerProbability = 0.2
 )
 
+var _ PeerTracker = (*peerTracker)(nil)
+
+type PeerTracker interface {
```

Comment:
introduce an interface to reuse this tracker in avalanchego. We should consider pulling this out from x package into avalanchego code and possibly reuse it more widely (althought we may want to use bandwidth rich nodes for critical processes, not any process)